src/backend/distributed/commands/alter_table.c

```diff
@@ -1572,3 +1577,76 @@ ExecuteQueryViaSPI(char *query, int SPIOK)
 		ereport(ERROR, (errmsg("could not finish SPI connection")));
 	}
 }
+
+
+/*
+ * SwitchToSequentialAndLocalExecutionIfRelationNameTooLong generates the longest shard name
+ * on the shards of a distributed table, and if exceeds the limit switches to sequential and
+ * local execution to prevent self-deadlocks.
+ *
+ * In case of a RENAME, the relation name parameter should store the new table name, so
+ * that the function can generate shard names of the renamed relations
+ */
+void
+SwitchToSequentialAndLocalExecutionIfRelationNameTooLong(Oid relationId,
+														 char *relationName)
+{
+	char *longestPartitionName = NULL;
+	if (!IsCitusTable(relationId))
+	{
+		return;
+	}
+
+	if (PartitionedTable(relationId))
+	{
+		longestPartitionName = LongestPartitionName(relationId);
+		if (longestPartitionName == NULL)
+		{
+			/* no partitions have been created yet */
+			return;
+		}
+	}
+
+	char *longestShardName = NULL;
+	if (longestPartitionName == NULL)
+	{
+		longestShardName = pstrdup(relationName);
+	}
+	else
+	{
+		longestShardName = pstrdup(relationName);
```

Comment:
the if and the else are the same here

Reply:
This was the result of an unsuccessful rebase. Fixed it now.